packages/youtube_player_iframe/lib/src/player_params.dart

```diff
@@ -129,6 +135,7 @@ class YoutubePlayerParams {
     this.mute = false,
     this.captionLanguage = 'en',
     this.enableCaption = true,
+    this.enablePointerEvents = true,
```

Comment:
Are there any consequences of turning this off by default ?

Reply:
@sarbagyastha Thanks for the reply

When `enablePointerEvents` is `false`, the `pointer-events` value is `none` in `player.html`, which is consistent with the value set in `raw_youtube_player.dart` 

By setting this to `false` as a default, it would mean that context menu in the iFrame can never be activated, personally if you ask me, should not have any side effects **unless** one finds the context menu useful

Reply:
Then let's make it `String? pointerEvents` to closely align with the CSS.
By default give it `pointer-events: initial` for html tag and `pointer-events: inherit` for all other tags. 

What do you think ?

Reply:
That's a good suggestion, but I think to make it less error prone and implement best code practises, we should create an enum class that covers all the `pointer-events` values as per [mozilla's documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events). Though on normal cases only values such as none, inherit, initial and auto, its better to just include everything for future-proofing

It should look something like this 

```
/// An enum for the set of pointer-events available to be passed in to `player.html`
enum PointerEvents {
  /// An enum for `pointer-events: auto`
  auto,

  /// An enum for `pointer-events: none`
  none,

  /// An enum for `pointer-events: inherit`
  inherit,

  /// An enum for `pointer-events: initial`
  initial,

  /// An enum for `pointer-events: revert`
  revert,

  /// An enum for `pointer-events: unset`
  unset,

  /// An enum for `pointer-events: revertLayer`
  revertLayer,
}

/// Function to get the string value of the pointer event based on the enum provided
String getPointerEvent(PointerEvents event) {
  switch (event) {
    case (PointerEvents.auto):
      {
        return 'auto';
      }
    case (PointerEvents.none):
      {
        return 'none';
      }
    case (PointerEvents.inherit):
      {
        return 'inherit';
      }
    case (PointerEvents.initial):
      {
        return 'initial';
      }
    case (PointerEvents.revert):
      {
        return 'revert';
      }
    case (PointerEvents.unset):
      {
        return 'unset';
      }
    case (PointerEvents.revertLayer):
      {
        return 'revert-layer';
      }
    default:
      return 'auto';
  }
}
```

We can then change the class constructor to something like this 
```
final PointerEvents pointerEvents
this.pointerEvents = PointerEvents.initial
```

Then in the `load()` function would look something like this 
```
Future<void> load({
    required YoutubePlayerParams params,
    String? baseUrl,
  }) async {
    final playerHtml = await rootBundle.loadString(
      'packages/youtube_player_iframe/assets/player.html',
    );

    final controller = await _webViewControllerCompleter.future;
    final platform = kIsWeb ? 'web' : defaultTargetPlatform.name.toLowerCase();
    final pointerEvents = params.enablePointerEvents ? 'auto' : 'none';

    await controller.loadHtmlString(
      playerHtml
          .replaceAll('<<pointerEvents>>', pointerEvents)
          .replaceFirst('<<playerVars>>', params.toJson())
          .replaceFirst('<<platform>>', platform)
          .replaceFirst('<<host>>', params.origin ?? 'https://www.youtube.com'),
      baseUrl: baseUrl,
    );
  }
```

The `player.html` would be as per your suggestions

Reply:
Sounds good, but let's use the new enum fields, instead of using extension.
Also, let's pass pointer events just to html tag and use `inherit` on other tags.